### DISCUSSION TITLE
iOS - why didn't ask for photo permission but can select photo with "one-off photo selection"?

### DISCUSSION THREAD

**Discussion by Xuhehehe99** [yasirkula/UnityNativeGallery#292] [Q&A]
![IMG_4698](https://github.com/yasirkula/UnityNativeGallery/assets/142370794/0687f928-0c62-4bf6-be05-b2a0954fa4a9)

I tried both manually request permission "NativeGallery.RequestPermission" and just internally request inside "NativeGallery.GetImageFromGallery" but when I select my button to pick a pic, there's no something pop up to ask for access to the photo, but can directly select one. And the Console says: "Permission result: Granted". After i found out in the photo privacy settings, my app didn't get access to photo library but can select image from it, as the screenshot shows. How can I make it ask for the permission first?

**Comment by yasirkula** [comment]
Permission dialog isn't needed. Apple won't cause any issues during app review either because the image picker is their own design. It's designed to work without requiring any permissions. You can modify the native source code to always ask permissions but then you'll have a permission dialog with "Full Photos", "Selected Photos Only" and "Deny" options and the option that the user chooses will have no effect on their user experience, so it's pointless to require a permission dialog.

**Reply by Xuhehehe99** [reply]
thanks a loooot!!!

**Comment by SachinWagde67** [comment]
@yasirkula Is this the same for Android also? After updating the package to v1.8.1, it does not show any pop-up for asking access to the photos and directly shows the popup for selecting the photos. Is this how it should work? asking because I got an email from Google Play regarding the new Photo and Videos Permission Policy. By directly showing the popup to select the photos, will this cause any issues?

**Reply by yasirkula** [reply]
Depends on the Android version: https://github.com/yasirkula/UnityNativeGallery?tab=readme-ov-file#c-runtime-permissions

Note that NativeGallery never asks the READ_MEDIA_IMAGES and READ_MEDIA_VIDEO permissions, just make sure that your plugin is up-to-date.

**Reply by SachinWagde67** [reply]
The plugin is updated to v1.8.1. In the AndroidManifest file, there is only one permission, WRITE_EXTERNAL_STORAGE.
After updating, I checked in **Android 14**, but the permission popup didn't come, it directly asked me to select the photos from the storage. Is this correct?
I also checked in **Android 7**, and it asked for permission first, cause of the above permission in the AndroidManifest file, which I want as my game's minimum API level is 24 and Target API level is 34.
Is this okay? Will this comply with Google's new permission policy?

One more question - In Google Mail, they said to [migrate to System photo picker](https://support.google.com/googleplay/android-developer/answer/14115180?hl=en#:~:text=Apps%20with%20one%2Dtime%20or%20infrequent%20use%20of%20photos%20are%20requested%20to%20remove%20READ_MEDIA_IMAGES%20and%C2%A0%20READ_MEDIA_VIDEO%20permissions%20from%20their%20app%20manifest%20and%20migrate%20to%20a%20system%20photo%20picker%20if%20necessary.). Is this System Photo Picker the same as the popup in which we can select the photos from the device storage? I'm a bit confused about this photo picker thing. If not, can you tell me what is this System Photo Picker?

**Reply by yasirkula** [reply]
The behaviour sounds correct:

> NativeGallery doesn't require any permissions for picking images/videos from Gallery on Android 34+.

I don't think ACTION_PICK and ACTION_GET_CONTENT Intents count as system photo picker but since they don't require the sensitive permissions, I don't see any issues.

I'm expecting no issues with Android and iOS permissions policies.

**Reply by SachinWagde67** [reply]
Thanks!!!!
